client/js/components/header/search.js

```diff
@@ -0,0 +1,43 @@
+import showHide from './../show-hide';
+import { KEYS } from './../../util';
+
+const headerSearch = (el) => {
+  const headerSearchInput = el.querySelector('.js-search-input');
+  const searchToggle = showHide({el: el});
+
+  const init = () => {
+    handleEvents();
+  };
+
+  const handleEvents = () => {
+    el.addEventListener('submit', (event) => {
+      if (searchToggle.getActive() && headerSearchInput.value.trim().length) return;
```

Comment:
Could we compare `headerSearchInput.value.trim().length` to a value? Using JavaScript magic always causes problems. I'd like to introduce typing at some stage on the server, which means we could get the benefits on the client too.